From a937ba2b9b6132883d727f856911ae31d22ad619 Mon Sep 17 00:00:00 2001
From: Aaron Kimball <aaron@cloudera.com>
Date: Fri, 12 Mar 2010 17:46:26 -0800
Subject: [PATCH 0206/1020] MAPREDUCE-1394. Sqoop generates incorrect URIs in paths sent to Hive

Description: Hive used to require a ':8020' in HDFS URIs used with LOAD DATA statements, even though the normalized form of such a URI does not contain an explicit port number (since 8020 is the default port). Sqoop matched this by hacking the URI strings it forwarded to Hive.

<p>Hive fixed this bug a while ago &#8211; Sqoop should catch up.</p>
Reason: bugfix (compatibility)
Author: Aaron Kimball
Ref: UNKNOWN
---
 .../apache/hadoop/sqoop/hive/TableDefWriter.java   |   17 -----------------
 1 files changed, 0 insertions(+), 17 deletions(-)

diff --git a/src/contrib/sqoop/src/java/org/apache/hadoop/sqoop/hive/TableDefWriter.java b/src/contrib/sqoop/src/java/org/apache/hadoop/sqoop/hive/TableDefWriter.java
index 81550af..ab8a9e6 100644
--- a/src/contrib/sqoop/src/java/org/apache/hadoop/sqoop/hive/TableDefWriter.java
+++ b/src/contrib/sqoop/src/java/org/apache/hadoop/sqoop/hive/TableDefWriter.java
@@ -143,23 +143,6 @@ public class TableDefWriter {
     FileSystem fs = FileSystem.get(configuration);
     Path finalPath = new Path(tablePath).makeQualified(fs);
     String finalPathStr = finalPath.toString();
-    if (finalPathStr.startsWith("hdfs://") && finalPathStr.indexOf(":", 7) == -1) {
-      // Hadoop removed the port number from the fully-qualified URL.
-      // We need to reinsert this or else Hive will complain.
-      // Do this right before the third instance of the '/' character.
-      int insertPoint = 0;
-      for (int i = 0; i < 3; i++) {
-        insertPoint = finalPathStr.indexOf("/", insertPoint + 1);
-      }
-
-      if (insertPoint == -1) {
-        LOG.warn("Fully-qualified HDFS path does not contain a port.");
-        LOG.warn("this may cause a Hive error.");
-      } else {
-        finalPathStr = finalPathStr.substring(0, insertPoint) + ":" + DEFAULT_HDFS_PORT
-            + finalPathStr.substring(insertPoint, finalPathStr.length());
-      }
-    }
 
     StringBuilder sb = new StringBuilder();
     sb.append("LOAD DATA INPATH '");
-- 
1.7.0.4

